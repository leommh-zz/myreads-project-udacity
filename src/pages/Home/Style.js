const style = {
    main: {
        backgroundColor: '#605872',
        padding: 20
    },
    title: {
        color: '#fff',
    },
    bookSearch: {
        backgroundColor: '#605872' 
    },
    add: {
        backgroundColor: '#fff',
        position: 'fixed',   
        bottom: 20,
        right: 20,

    },
}

export default style;