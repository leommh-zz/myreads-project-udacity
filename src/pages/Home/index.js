import React from 'react';
import { Link } from 'react-router-dom';
import { Grid, Button } from '@material-ui/core';
import Add from '@material-ui/icons/Add';
import BookCase from '../../components/BookCase';
import Loader from '../../components/Loader';
import If from '../../components/If';
import Style from './Style';

const Home = ({ booksUser, defineStatus, loading }) => (

    <Grid container style={Style.main}>
        {/* Loading Screen */}
        <If test={loading === true}>
          <Loader />
        </If>

        {/* Books Screen */}
        <If test={loading === false}>
            <Grid item style={Style.bookCase}>
                <BookCase title="Currently Reading" books={booksUser.currentlyReading} empty={false} defineStatus={defineStatus} />
                <BookCase title="Want to Read" books={booksUser.wantToRead} empty={false} defineStatus={defineStatus} />
                <BookCase title="Read" books={booksUser.read} empty={false} defineStatus={defineStatus} />
            </Grid>

            <Grid item container justify="flex-end" style={Style.openSearch}>
                <Link to="/search" className="add-contact">
                    <Button variant="fab" aria-label="Close" style={Style.add}>
                        <Add />
                    </Button>
                </Link>
            </Grid>
        </If>

    </Grid>
)

export default Home;