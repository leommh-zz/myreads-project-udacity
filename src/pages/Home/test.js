import React from 'react';
import { shallow } from 'enzyme';
import Home from './index';

describe('[Component] Home', () => {
    const booksUser = [
        {
            title: "Title Book",
            description: "This book and the table",
            categories: [
                "action"
            ],
            authors: [
                "Leonardo Morini"
            ],
            shelf: '',
            averageRating: 2
        }
    ]

    const defineStatus = () => console.log('Updated');
    const loading = false;

    const wrapper = shallow(
        <Home 
            booksUser={booksUser} 
            loading={loading} 
            defineStatus={defineStatus} 
        />
    );

    it('Display Home Component with props', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});