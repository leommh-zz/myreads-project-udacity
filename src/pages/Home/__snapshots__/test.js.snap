// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Home Display Home Component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    booksUser={
        Array [
            Object {
              "authors": Array [
                "Leonardo Morini",
              ],
              "averageRating": 2,
              "categories": Array [
                "action",
              ],
              "description": "This book and the table",
              "shelf": "",
              "title": "Title Book",
            },
          ]
    }
    defineStatus={[Function]}
    loading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <If
          test={false}
>
          <Loader />
</If>,
        <If
          test={true}
>
          <WithStyles(Grid)
                    item={true}
                    style={undefined}
          >
                    <BookCase
                              books={undefined}
                              defineStatus={[Function]}
                              empty={false}
                              title="Currently Reading"
                    />
                    <BookCase
                              books={undefined}
                              defineStatus={[Function]}
                              empty={false}
                              title="Want to Read"
                    />
                    <BookCase
                              books={undefined}
                              defineStatus={[Function]}
                              empty={false}
                              title="Read"
                    />
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    container={true}
                    item={true}
                    justify="flex-end"
                    style={undefined}
          >
                    <Link
                              className="add-contact"
                              replace={false}
                              to="/search"
                    >
                              <WithStyles(Button)
                                        aria-label="Close"
                                        style={
                                                  Object {
                                                            "backgroundColor": "#fff",
                                                            "bottom": 20,
                                                            "position": "fixed",
                                                            "right": 20,
                                                          }
                                        }
                                        variant="fab"
                              >
                                        <pure(AddIcon) />
                              </WithStyles(Button)>
                    </Link>
          </WithStyles(Grid)>
</If>,
      ],
      "container": true,
      "style": Object {
        "backgroundColor": "#605872",
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Loader />,
          "test": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
              style={undefined}
>
              <BookCase
                            books={undefined}
                            defineStatus={[Function]}
                            empty={false}
                            title="Currently Reading"
              />
              <BookCase
                            books={undefined}
                            defineStatus={[Function]}
                            empty={false}
                            title="Want to Read"
              />
              <BookCase
                            books={undefined}
                            defineStatus={[Function]}
                            empty={false}
                            title="Read"
              />
</WithStyles(Grid)>,
            <WithStyles(Grid)
              container={true}
              item={true}
              justify="flex-end"
              style={undefined}
>
              <Link
                            className="add-contact"
                            replace={false}
                            to="/search"
              >
                            <WithStyles(Button)
                                          aria-label="Close"
                                          style={
                                                        Object {
                                                                      "backgroundColor": "#fff",
                                                                      "bottom": 20,
                                                                      "position": "fixed",
                                                                      "right": 20,
                                                                    }
                                          }
                                          variant="fab"
                            >
                                          <pure(AddIcon) />
                            </WithStyles(Button)>
              </Link>
</WithStyles(Grid)>,
          ],
          "test": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <BookCase
                  books={undefined}
                  defineStatus={[Function]}
                  empty={false}
                  title="Currently Reading"
/>,
                <BookCase
                  books={undefined}
                  defineStatus={[Function]}
                  empty={false}
                  title="Want to Read"
/>,
                <BookCase
                  books={undefined}
                  defineStatus={[Function]}
                  empty={false}
                  title="Read"
/>,
              ],
              "item": true,
              "style": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "books": undefined,
                  "defineStatus": [Function],
                  "empty": false,
                  "title": "Currently Reading",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "books": undefined,
                  "defineStatus": [Function],
                  "empty": false,
                  "title": "Want to Read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "books": undefined,
                  "defineStatus": [Function],
                  "empty": false,
                  "title": "Read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                className="add-contact"
                replace={false}
                to="/search"
>
                <WithStyles(Button)
                                aria-label="Close"
                                style={
                                                Object {
                                                                "backgroundColor": "#fff",
                                                                "bottom": 20,
                                                                "position": "fixed",
                                                                "right": 20,
                                                              }
                                }
                                variant="fab"
                >
                                <pure(AddIcon) />
                </WithStyles(Button)>
</Link>,
              "container": true,
              "item": true,
              "justify": "flex-end",
              "style": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  aria-label="Close"
                  style={
                                    Object {
                                                      "backgroundColor": "#fff",
                                                      "bottom": 20,
                                                      "position": "fixed",
                                                      "right": 20,
                                                    }
                  }
                  variant="fab"
>
                  <pure(AddIcon) />
</WithStyles(Button)>,
                "className": "add-contact",
                "replace": false,
                "to": "/search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "Close",
                  "children": <pure(AddIcon) />,
                  "style": Object {
                    "backgroundColor": "#fff",
                    "bottom": 20,
                    "position": "fixed",
                    "right": 20,
                  },
                  "variant": "fab",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <If
            test={false}
>
            <Loader />
</If>,
          <If
            test={true}
>
            <WithStyles(Grid)
                        item={true}
                        style={undefined}
            >
                        <BookCase
                                    books={undefined}
                                    defineStatus={[Function]}
                                    empty={false}
                                    title="Currently Reading"
                        />
                        <BookCase
                                    books={undefined}
                                    defineStatus={[Function]}
                                    empty={false}
                                    title="Want to Read"
                        />
                        <BookCase
                                    books={undefined}
                                    defineStatus={[Function]}
                                    empty={false}
                                    title="Read"
                        />
            </WithStyles(Grid)>
            <WithStyles(Grid)
                        container={true}
                        item={true}
                        justify="flex-end"
                        style={undefined}
            >
                        <Link
                                    className="add-contact"
                                    replace={false}
                                    to="/search"
                        >
                                    <WithStyles(Button)
                                                aria-label="Close"
                                                style={
                                                            Object {
                                                                        "backgroundColor": "#fff",
                                                                        "bottom": 20,
                                                                        "position": "fixed",
                                                                        "right": 20,
                                                                      }
                                                }
                                                variant="fab"
                                    >
                                                <pure(AddIcon) />
                                    </WithStyles(Button)>
                        </Link>
            </WithStyles(Grid)>
</If>,
        ],
        "container": true,
        "style": Object {
          "backgroundColor": "#605872",
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Loader />,
            "test": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
                style={undefined}
>
                <BookCase
                                books={undefined}
                                defineStatus={[Function]}
                                empty={false}
                                title="Currently Reading"
                />
                <BookCase
                                books={undefined}
                                defineStatus={[Function]}
                                empty={false}
                                title="Want to Read"
                />
                <BookCase
                                books={undefined}
                                defineStatus={[Function]}
                                empty={false}
                                title="Read"
                />
</WithStyles(Grid)>,
              <WithStyles(Grid)
                container={true}
                item={true}
                justify="flex-end"
                style={undefined}
>
                <Link
                                className="add-contact"
                                replace={false}
                                to="/search"
                >
                                <WithStyles(Button)
                                                aria-label="Close"
                                                style={
                                                                Object {
                                                                                "backgroundColor": "#fff",
                                                                                "bottom": 20,
                                                                                "position": "fixed",
                                                                                "right": 20,
                                                                              }
                                                }
                                                variant="fab"
                                >
                                                <pure(AddIcon) />
                                </WithStyles(Button)>
                </Link>
</WithStyles(Grid)>,
            ],
            "test": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <BookCase
                    books={undefined}
                    defineStatus={[Function]}
                    empty={false}
                    title="Currently Reading"
/>,
                  <BookCase
                    books={undefined}
                    defineStatus={[Function]}
                    empty={false}
                    title="Want to Read"
/>,
                  <BookCase
                    books={undefined}
                    defineStatus={[Function]}
                    empty={false}
                    title="Read"
/>,
                ],
                "item": true,
                "style": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "books": undefined,
                    "defineStatus": [Function],
                    "empty": false,
                    "title": "Currently Reading",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "books": undefined,
                    "defineStatus": [Function],
                    "empty": false,
                    "title": "Want to Read",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "books": undefined,
                    "defineStatus": [Function],
                    "empty": false,
                    "title": "Read",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  className="add-contact"
                  replace={false}
                  to="/search"
>
                  <WithStyles(Button)
                                    aria-label="Close"
                                    style={
                                                      Object {
                                                                        "backgroundColor": "#fff",
                                                                        "bottom": 20,
                                                                        "position": "fixed",
                                                                        "right": 20,
                                                                      }
                                    }
                                    variant="fab"
                  >
                                    <pure(AddIcon) />
                  </WithStyles(Button)>
</Link>,
                "container": true,
                "item": true,
                "justify": "flex-end",
                "style": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    aria-label="Close"
                    style={
                                        Object {
                                                            "backgroundColor": "#fff",
                                                            "bottom": 20,
                                                            "position": "fixed",
                                                            "right": 20,
                                                          }
                    }
                    variant="fab"
>
                    <pure(AddIcon) />
</WithStyles(Button)>,
                  "className": "add-contact",
                  "replace": false,
                  "to": "/search",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "Close",
                    "children": <pure(AddIcon) />,
                    "style": Object {
                      "backgroundColor": "#fff",
                      "bottom": 20,
                      "position": "fixed",
                      "right": 20,
                    },
                    "variant": "fab",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
